.github/workflows/ci.yaml

```diff
@@ -0,0 +1,47 @@
+name: ci
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    # The type of runner that the job will run on
```

Comment:
I don't think we need an annotated config. Comments should only explain motivation and weird corner cases.